src/main/java/com/bandwidth/sqs/consumer/acknowledger/MessageAcknowledger.java

```diff
@@ -111,28 +87,31 @@ public void delay(Duration newVisibilityTimeout) {
      * It may be a good idea to modify the message and include your own retry count, since the SQS receiveCount will be
      * reset to 0 with a new message.
      */
-    public void replace(T newMessage, Duration delay) {
+    public void replace(T newMessage, Optional<Duration> delay) {
         ackModeSingle.onSuccess(AckMode.MODIFY);
-        doTransfer(newMessage, queueUrl, delay).subscribeWith(ackingComplete);
+        doTransfer(newMessage, sqsQueue, delay).subscribeWith(ackingComplete);
+    }
+
+    public void replace(T newMessage) {
+        replace(newMessage, Optional.empty());
     }
 
     /**
      * Same as replace, but the modified message is published to a different SQS queue.
      * If you want to publish a modified or delayed message to the SAME queue, you must use replace or requeue instead.
      */
-    public void transfer(T newMessage, String newQueueUrl, Duration delay) {
+    public void transfer(T newMessage, SqsQueue<T> newQueue, Optional<Duration> delay) {
         ackModeSingle.onSuccess(AckMode.TRANSFER);
-        doTransfer(newMessage, newQueueUrl, delay).subscribeWith(ackingComplete);
+        doTransfer(newMessage, newQueue, delay).subscribeWith(ackingComplete);
     }
 
-    private Completable doTransfer(T newMessage, String newQueueUrl, Duration delay) {
-        DeleteMessageRequest deleteRequest = new DeleteMessageRequest()
-                .withQueueUrl(queueUrl)
-                .withReceiptHandle(receiptId);
+    public void transfer(T newMessage, SqsQueue<T> newQueue){
+        transfer(newMessage, newQueue, Optional.empty());
+    }
 
-        return messagePublisher.publishMessage(newMessage, newQueueUrl, Optional.of(delay))
-                .flatMap((response) -> sqsClient.deleteMessage(deleteRequest))
-                .toCompletable();
+    private Completable doTransfer(T newMessage, SqsQueue<T> newQueue, Optional<Duration> delay) {
```

Comment:
New is needed / makes sense here. It is transferring between 2 queues, and you need to distinguish between the current (old) queue, and the new one.